python/ray/data/_internal/numpy_support.py

```diff
@@ -86,8 +86,11 @@ def convert_udf_returns_to_numpy(udf_return_col: Any) -> Any:
                 elif not np.isscalar(e):
                     has_object = True
             if has_object or len(shapes) > 1:
-                # This util works around some limitations of np.array(dtype=object).
-                udf_return_col = create_ragged_ndarray(udf_return_col)
+                if isinstance(e, list) and all(isinstance(val, str) for val in e):
+                    udf_return_col = np.array(udf_return_col)
```

Comment:
I took a look around the code base and it didn't seem like `create_ragged_ndarray` was intended for lists of all `strings`.

Reply:
we shouldn't need to implement a specific case for string arrays here i think -- does using `_create_possibly_ragged_ndarray()` method here make sense? This method will try to first construct the array with `np.array(values, copy=False)`, then falls back to `create_ragged_ndarray()`.

Reply:
This works!